catalog/src/main/java/it/eng/catalog/exceptions/CatalogExceptionAdvice.java

```diff
@@ -14,28 +15,29 @@
 import it.eng.catalog.model.Reason;
 import it.eng.catalog.model.Serializer;
 import it.eng.catalog.rest.protocol.CatalogProtocolController;
-import it.eng.catalog.service.CatalogService;
 import jakarta.validation.ValidationException;
 
-@RestControllerAdvice(basePackageClasses = {CatalogProtocolController.class, CatalogService.class})
+@RestControllerAdvice(basePackageClasses = CatalogProtocolController.class)
 public class CatalogExceptionAdvice extends ResponseEntityExceptionHandler {
 
     @ExceptionHandler(value = {CatalogErrorException.class})
-    protected ResponseEntity<String> handleCatalogErrorException(CatalogErrorException ex) {
+    protected ResponseEntity<Object> handleCatalogErrorException(CatalogErrorException ex) {
         CatalogError catalogError = CatalogError.Builder.newInstance()
         		.code(HttpStatus.NOT_FOUND.getReasonPhrase())
                 .reason(Arrays.asList(Reason.Builder.newInstance().language("en").value(ex.getLocalizedMessage()).build()))
                 .build();
+//        return handleExceptionInternal(ex, response, new HttpHeaders(), HttpStatus.NOT_FOUND, request);
```

Comment:
handleExceptionInternal or  return new ResponseEntity<>?

Reply:
yep, we agreed to use handleExceptionInternal, I forgot to change it here too, good that you found it :)